src/libraries/System.Transactions.Local/src/System/Transactions/Oletx/OletxTransactionManager.cs

```diff
@@ -0,0 +1,809 @@
+                    {
+                        localProxyShimFactory.GetNotification(
+                            out enlistment2,
+                            out shimNotificationType,
+                            out isSinglePhase,
+                            out abortingHint,
+                            out holdingNotificationLock,
+                            out prepareInfoBuffer);
+                    }
+                    finally
+                    {
+                        if (holdingNotificationLock)
+                        {
+                            if (enlistment2 is OletxInternalResourceManager)
+                            {
+                                // In this case we know that the TM has gone down and we need to exchange
+                                // the native lock for a managed lock.
+                                ProcessingTmDown = true;
+#pragma warning disable 0618
+                                //@TODO: This overload of Monitor.Enter is obsolete.  Please change this to use Monitor.Enter(ref bool), and remove the pragmas   -- ericeil
+                                Monitor.Enter(ProxyShimFactory);
+#pragma warning restore 0618
+                            }
+                            else
+                            {
+                                holdingNotificationLock = false;
+                            }
+                            localProxyShimFactory.ReleaseNotificationLock();
+                        }
+                        Thread.EndThreadAffinity();
+                    }
+
+                    // If a TM down is being processed it is possible that the native lock
+                    // has been exchanged for a managed lock.  In that case we need to attempt
+                    // to take a lock to hold up processing more events until the TM down
+                    // processing is complete.
+                    if (ProcessingTmDown)
+                    {
+                        lock (ProxyShimFactory)
+                        {
+                            // We don't do any work under this lock just make sure that we
+                            // can take it.
+                        }
+                    }
+
+                    if (shimNotificationType != ShimNotificationType.None)
+                    {
+                        // Next, based on the notification type, cast the Handle accordingly and make
+                        // the appropriate call on the enlistment.
+                        switch (shimNotificationType)
+                        {
+                            case ShimNotificationType.Phase0RequestNotify:
+                                {
+                                    if (enlistment2 is OletxPhase0VolatileEnlistmentContainer ph0VolEnlistContainer)
+                                    {
+                                        //CSDMain 91509 - We now synchronize this call with the AddDependentClone call in RealOleTxTransaction
```

Comment:
Are these bug databases still accessible?  Wondering if such comments can be removed.

Reply:
I honestly don't know... I went through an adventure of trying to find an old bug database for https://github.com/dotnet/runtime/issues/63630, but didn't manage to... If you think it's important I can make a note and try to work on it.

Reply:
It's not important.  But if it's not meaningful to us now, it's not likely to become more meaningful in the future :), nor is it valuable for open source even if we can find the database internally.

Reply:
OK then, I'll remove these.